sycl/test-e2e/Annotated_arg_ptr/annotated_ptr.cpp

```diff
@@ -28,6 +28,12 @@ int main() {
 
   auto *d = malloc_shared<int>(4, Q);
   auto d_ptr = annotated_ptr{d};
+
+  auto *e = malloc_shared<int>(1, Q);
+  volatile int *e_vol = e;
+  *e_vol = 0;
+  auto e_ptr = annotated_ptr{e_vol};
```

Comment:
Could you also please add a `const volatile` case?

Reply:
done